src/hoc/withTheme.js

```diff
@@ -3,34 +3,49 @@
 
 import React from 'react'
 import PropTypes from 'prop-types'
+import hoistStatics from 'hoist-non-react-statics'
 import { CHANNEL } from '../models/ThemeProvider'
 
-export default (Component: ReactClass<mixed>) => class extends React.Component {
-  static contextTypes = {
-    [CHANNEL]: PropTypes.func,
-  };
+const wrapWithTheme = (Component: ReactClass<any>) => {
+  const componentName = (
+    Component.displayName ||
+    Component.name ||
+    'Component'
+  )
 
-  state: { theme?: ?Object } = {};
-  unsubscribe: () => void;
+  class WithTheme extends React.Component {
```

Comment:
That's not a good idea, since we'd be overriding some original logic. I'd stick to the HOC-pattern :wink: